tests/testthat/test-reporter-list.R

```diff
@@ -1,25 +1,59 @@
+# regression test: test_file() used to crash with a NULL reporter
+test_that("ListReporter with test_file and NULL reporter", {
+  test_file_path <- 'test-list-reporter/test-exercise-list-reporter.R'
+  expect_error(test_file(test_path(test_file_path), reporter = NULL), NA)
+})
+
 test_that("ListReporter with test_file", {
```

Comment:
This is a very large test. Do you think you could break it up into smaller pieces?

Reply:
How so ? The test is ~ 24 lines, and has a single call to `test_file()`.

Reply:
It's sufficiently long that I can't follow exactly what it's doing.